DiscordAlertsTrader/brokerages/ibkr_api.py

```diff
@@ -0,0 +1,248 @@
+                'filledQuantity': order.filledQuantity,
+                'price':trade.orderStatus.avgFillPrice,
+                'orderStrategyType': 'SINGLE',
+                "order_id" : order.orderId,
+                "orderId": order.orderId,
+                "stopPrice": order.auxPrice if order.orderType == 'STP' else None,
+                'orderType':  order.orderType,
+                'placedTime': trade.log[0].time if trade.log else int(time.time()*1000),
+                'enteredTime': trade.fills[0].time if trade.fills else int(time.time()*1000),
+                "closeTime": trade.fills[-1].time if trade.fills else int(time.time()*1000),
+                'orderLegCollection':[{
+                    'instrument':{'symbol':trade.contract.symbol},
+                    'instruction': order.action,
+                    'quantity': order.filledQuantity,
+                }]             
+
+            }
+            formatted_order = self.format_order(order_info)
+            orders_inf.append(formatted_order)
+
+        acc_inf['securitiesAccount']['orderStrategies'] = orders_inf
+        # print(acc_inf)
+        return acc_inf
+
+    def format_order(self, order:dict):
+        """ output format for order_response.Order, mimicks the order_info from TDA API"""
+        stopPrice= order['stopPrice']
+        timestamp = order['placedTime']/1000 if order['placedTime'] else None
+        enteredTime = datetime.fromtimestamp(timestamp).strftime("%Y-%m-%dT%H:%M:%S+00") if timestamp else None
+        if 'closeTime' in order:
+            timestamp = order['closeTime']/1000 if order['closeTime'] else None
+            closeTime = datetime.fromtimestamp(timestamp).strftime("%Y-%m-%dT%H:%M:%S+00") if timestamp else None
+        else:
+            closeTime = enteredTime
+        status = order['status'].upper().replace('FILLED','FILLED').replace('SUBMITTED','WORKING').replace('CANCELLED','CANCELED')
+        order_info = {
+            'status': status,
+            'quantity': order['quantity'],
+            'filledQuantity': order['filledQuantity'],
+            'price':order['price'],
+            'orderStrategyType': 'SINGLE',
+            "order_id" : order['orderId'],
+            "orderId": order['orderId'],
+            "stopPrice": stopPrice if stopPrice else None,
+            'orderType':  order['orderType'],
+            'enteredTime': enteredTime,
+            "closeTime": closeTime,
+            'orderLegCollection':[{
+                'instrument':{'symbol':order['orderLegCollection'][0]['instrument']['symbol']},
+                'instruction': order['orderLegCollection'][0]['instruction'],
+                'quantity': order['orderLegCollection'][0]['quantity'],
+            }]             
+        }
+        return order_info
+    
+    def get_quotes(self, symbol:list):
+        pass
+    def send_order(self, side:str, order_type:str, quantity:int, contract: dict, price=None, stop_price=None):
```

Comment:
Could you add functions that generate contracts that can be read by `send_order` with bto lims, stc lim, bracket orders and trailing stops, like these:

    def make_BTO_lim_order(self, Symbol:str, Qty:int, price:float, action="BTO", **kwarg):
        "Buy with a limit order"

        kwargs = {}
        if action == "BTO":
            kwargs['action'] = "BUY"
        elif action == "STO":
            print("STO not available for WeBull")
            return
        Symbol = self.fix_symbol(Symbol, "in")
        if "_" in Symbol:
            kwargs['asset'] = 'option'
            optionId = self.get_option_id(Symbol)
            if optionId is None:
                print("No optionId found for", Symbol)
                return None        
            kwargs['optionId'] = optionId
        else:
            kwargs['asset'] ='stock'
            kwargs['outsideRegularTradingHour'] = True
            kwargs['stock'] = Symbol

        kwargs['enforce'] ='GTC'
        kwargs['quant'] = Qty  
        kwargs['orderType'] = 'LMT' 
        kwargs['lmtPrice'] = price
        return kwargs

    def make_Lim_SL_order(self, Symbol:str, Qty:int,  PT:float, SL:None, action="STC", **kwarg):
        """Sell with a limit order and a stop loss order"""        
        kwargs = {}
        if action == "STC":
            kwargs['action'] = "SELL"
        elif action == "BTC":
            print("BTC not available for WeBull")
            return

        Symbol = self.fix_symbol(Symbol, "in")
        if "_" in Symbol:
            kwargs['asset'] = 'option'
            optionId = self.get_option_id(Symbol)
            if optionId is None:
                print("No optionId found for", Symbol)
                return None        
            kwargs['optionId'] = optionId
        else:
            kwargs['asset'] ='stock'
            kwargs['outsideRegularTradingHour'] = True
            kwargs['stock'] = Symbol

        kwargs['enforce'] ='GTC'
        kwargs['quant'] = Qty 
        if SL is not None:
            print("WARNING: webull api does not support OCO, setting a SL only, do not provide SL to send PT")
            kwargs['orderType'] = 'STP' 
            kwargs['lmtPrice'] = SL
        else:
            print("WARNING: webull api does not support OCO, sending PT without SL")
            kwargs['orderType'] = 'LMT' 
            kwargs['lmtPrice'] = PT
        return kwargs

    def make_STC_lim(self, Symbol:str, Qty:int, price:float, strike=None, action="STC", **kwarg):
        """Sell with a limit order and a stop loss order"""        
        kwargs = {}
        if action == "STC":
            kwargs['action'] = "SELL"
        elif action == "BTC":
            print("BTC not available for WeBull")
            return

        Symbol = self.fix_symbol(Symbol, "in")
        if "_" in Symbol:
            kwargs['asset'] = 'option'
            optionId = self.get_option_id(Symbol)
            if optionId is None:
                print("No optionId found for", Symbol)
                return None        
            kwargs['optionId'] = optionId
        else:
            kwargs['asset'] ='stock'
            kwargs['outsideRegularTradingHour'] = True
            kwargs['stock'] = Symbol

        kwargs['enforce'] ='GTC'
        kwargs['quant'] = Qty  
        kwargs['orderType'] = 'LMT' 
        kwargs['lmtPrice'] = price
        return kwargs

    def make_STC_SL(self, Symbol:str, Qty:int, SL:float, action="STC", **kwarg):
        """Sell with a stop loss order"""
        kwargs = {}
        if action == "STC":
            kwargs['action'] = "SELL"
        elif action == "BTC":
            print("BTC not available for WeBull")
            return

        Symbol = self.fix_symbol(Symbol, "in")
        if "_" in Symbol:
            kwargs['asset'] = 'option'
            optionId = self.get_option_id(Symbol)
            if optionId is None:
                print("No optionId found for", Symbol)
                return None        
            kwargs['optionId'] = optionId
        else:
            kwargs['asset'] ='stock'
            kwargs['outsideRegularTradingHour'] = False
            kwargs['stock'] = Symbol

        kwargs['enforce'] ='GTC'
        kwargs['quant'] = Qty 
        kwargs['orderType'] = 'STP' 
        kwargs['stpPrice'] = SL
        kwargs['lmtPrice'] = SL
        return kwargs

    def make_STC_SL_trailstop(self, Symbol:str, Qty:int,  trail_stop_const:float, action="STC", **kwarg):
        "trail_stop_const"
        kwargs = {}
        if action == "STC":
            kwargs['action'] = "SELL"
        elif action == "BTC":
            print("BTC not available for WeBull")
            return

        Symbol = self.fix_symbol(Symbol, "in")
        if "_" in Symbol:
            print("WARNING webull does not support trailing stop for options")
            return {}        

        kwargs['asset'] ='stock'
        kwargs['outsideRegularTradingHour'] = True
        kwargs['stock'] = Symbol
        kwargs['enforce'] ='GTC'
        kwargs['quant'] = Qty 
        kwargs['orderType'] = 'STP TRAIL'
        kwargs['trial_value'] = trail_stop_const
        kwargs['trial_type'] = 'DOLLAR'
        kwargs['outsideRegularTradingHour'] = True
        return kwargs